stdlib/public/core/SmallString.swift

```diff
@@ -241,22 +241,37 @@ extension _SmallString {
   fileprivate mutating func withMutableCapacity(
     _ f: (UnsafeMutableRawBufferPointer) throws -> Int
   ) rethrows {
-    let len = try withUnsafeMutableBytes(of: &self._storage) {
-      (rawBufPtr: UnsafeMutableRawBufferPointer) -> Int in
-      let len = try f(rawBufPtr)
-      UnsafeMutableRawBufferPointer(
-        rebasing: rawBufPtr[len...]
-      ).initializeMemory(as: UInt8.self, repeating: 0)
-      return len
-    }
-    if len == 0 {
+    let len = try withUnsafeMutableBytes(of: &_storage, f)
+
+    if len <= 0 {
+      _debugPrecondition(len == 0)
       self = _SmallString()
       return
     }
-    _internalInvariant(len <= _SmallString.capacity)
+    zeroTrailingBytes(of: &_storage, from: len)
+    self = _SmallString(leading: _storage.0, trailing: _storage.1, count: len)
+  }
+}
 
-    let (leading, trailing) = self.zeroTerminatedRawCodeUnits
-    self = _SmallString(leading: leading, trailing: trailing, count: len)
+@inlinable @inline(__always)
+internal func zeroTrailingBytes(
+  of storage: inout _SmallString.RawBitPattern, from len: Int
```

Comment:
Nit: I think in the stdlib we usually put each parameter on a separate line when we use this style of declaration; obviously not obligatory, but if someone's used to reading that style then they may totally miss the second parameter here particularly since it's so short.

```suggestion
  of storage: inout _SmallString.RawBitPattern,
  from len: Int
```

Reply:
I disagree on the line break; see line 96 of `docs/StandardLibraryProgrammersManual.md`. Fair point on the name `len`, though; I'm changing it to `index` along with the detail of adding an underscore to `zeroTrailingbytes()`.